concent_api/gatekeeper/tests/test_gatekeeper_views.py

```diff
@@ -0,0 +1,171 @@
+import datetime
+from base64         import b64encode
+
+from freezegun      import freeze_time
+from django.conf    import settings
+from django.http    import JsonResponse
+from django.test    import Client
+from django.test    import override_settings
+from django.test    import TestCase
+from django.urls    import reverse
+
+from golem_messages.shortcuts       import dump
+from golem_messages.message         import FileTransferToken
+
+
+@override_settings(
+    CONCENT_PRIVATE_KEY = b'l\xcdh\x19\xeb$>\xbcG\xa1\xc7v\xe8\xd7o\x0c\xbf\x0e\x0fM\x89lw\x1e\xd7K\xd6Hnv$\xa2',
+    CONCENT_PUBLIC_KEY  = b'\xf3\x97\x19\xcdX\xda\x86tiP\x1c&\xd39M\x9e\xa4\xddb\x89\xb5,]O\xd5cR\x84\xb85\xed\xc9\xa17e,\xb2s\xeb\n1\xcaN.l\xba\xc3\xb7\xc2\xba\xff\xabN\xde\xb3\x0b\xa6l\xbf6o\x81\xe0;',
+    STORAGE_CLUSTER_ADDRESS = 'http://devel.concent.golem.network/'
+)
+class GatekeeperViewUploadTest(TestCase):
+
+    @freeze_time("2018-12-30 11:00:00")
+    def setUp(self):
+        self.message_timestamp  = int(datetime.datetime.now().timestamp())
+        self.public_key         = '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=='
+        self.upload_token                                 = FileTransferToken()
+        self.upload_token.token_expiration_deadline       = self.message_timestamp + 3600
+        self.upload_token.storage_cluster_address         = 'http://devel.concent.golem.network/'
+        self.upload_token.authorized_client_public_key    = b'\xf3\x97\x19\xcdX\xda\x86tiP\x1c&\xd39M\x9e\xa4\xddb\x89\xb5,]O\xd5cR\x84\xb85\xed\xc9\xa17e,\xb2s\xeb\n1\xcaN.l\xba\xc3\xb7\xc2\xba\xff\xabN\xde\xb3\x0b\xa6l\xbf6o\x81\xe0;'
+
+        self.upload_token.files                 = [FileTransferToken.FileInfo()]
+        self.upload_token.files[0]['path']      = 'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend'
+        self.upload_token.files[0]['checksum']  = '098f6bcd4621d373cade4e832627b4f6'
+        self.upload_token.files[0]['size']      = 1024
+        self.upload_token.operation             = 'upload'
+
+    @freeze_time("2018-12-30 11:00:00")
+    def test_upload_should_accept_valid_message(self):
+        golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response = Client().post(
```

Comment:
You don't have to create a new `Client()` instance yourself. It already exists. Django creates `self.client` for you automatically.